File: tp_5_1_internet/app/src/main/java/com/example/tp_5_1_internet/model/Todo.java
package com.example.tp_5_1_internet.model;
import org.json.JSONObject;

public class Todo {

    private int userId;
    private int id;
    private String title;
    private boolean completed;

    public Todo(JSONObject jsonObject) {
        this.userId = jsonObject.optInt("userId");
        this.id = jsonObject.optInt("id");
        this.title = jsonObject.optString("title");
        this.completed = jsonObject.optBoolean("completed");
    }

    @Override
    public String toString() {
        return "Todo{" +
                "userId=" + userId +
                ", id=" + id +
                ", title='" + title + '\'' +
                ", completed=" + completed +
                '}';
    }

    public int getUserId() {
        return userId;
    }

    public void setUserId(int userId) {
        this.userId = userId;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public boolean isCompleted() {
        return completed;
    }

    public void setCompleted(boolean completed) {
        this.completed = completed;
    }
}

File: tp_5_1_internet_kotlin/app/src/main/java/com/example/tp_5_1_internet_kotlin/model/Todo.kt
package com.example.tp_5_1_internet_kotlin.model

import org.json.JSONObject

class Todo(jsonObject: JSONObject) {

    private val id: Int
    private val userId: Int
    private val title: String
    private val completed: Boolean

    init {
       with(jsonObject) {
           id = optInt("userId")
           userId = optInt("id")
           title = optString("title")
           completed = optBoolean("completed")
       }
    }

    override fun toString(): String {
        return "Todo(id=$id, userId=$userId, title='$title', completed=$completed)"
    }

}